### PR TITLE
build(android): Bump sentry-android to 5.0.0

### PR DESCRIPTION
**Pull request by jennmueng**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring

## :scroll: Description

Bumps Sentry Android SDK to `5.0.0`. AFAIK there should not be any change needed to RN.

## :green_heart: How did you test it?
Through Android sample app, and will see e2e tests on this PR.

## :pencil: Checklist

- [x] I reviewed submitted code
- [ ] I added tests to verify changes
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

**Review by bruno-garcia — APPROVED** [review]
Should be fine but @marandaneto will know for sure

**Review by philipphofmann — APPROVED** [review]
Uh nice.

File: android/build.gradle [hunk 1]
```diff
@@ -26,5 +26,5 @@ android {
 
 dependencies {
     implementation 'com.facebook.react:react-native:+'
-    api 'io.sentry:sentry-android:5.0.0-beta.4'
+    api 'io.sentry:sentry-android:5.0.0'
```
Comment (marandaneto) [inline android/build.gradle:29]: it'd be nice to address the breaking changes too, see https://github.com/getsentry/sentry-dart/pull/486/files
eg renaming `cacheDirSize` to `maxCacheItems`

Comment (jennmueng) [reply android/build.gradle:29]: RN didn't support the `cacheDirSize` option yet, which I guess is a downside of having to set each option individually when we have to maintain multiple hybrid SDKs... On a related note but off topic from this PR, @bruno-garcia suggested we should take a map for options from hybrid Sdks

Comment (marandaneto) [reply android/build.gradle:29]: yep, indeed, this is blocked by https://github.com/getsentry/sentry-java/issues/1117 though
so all good for now

**Review by marandaneto — APPROVED** [review]

**Pull request merged**